docs/frontend-system/building-plugins/06-swappable-components.md

```diff
@@ -0,0 +1,98 @@
+---
+id: swappable-components
+title: Swappable components
+sidebar_label: Swappable components
+description: Configuring or overriding Swappable Components
+---
+
+# Swappable components
+
+Swappable components are a feature of the frontend system that allow you to replace the implementations of components that are used in your Backstage app.
+These Swappable Components are defined using `createSwappableComponent` and then can be exported from a plugins `-react` package in order to be used in both other plugins, and to be rebound to a new implementation by the Backstage Integrator.
+
+## Creating a Swappable Component
+
+In order to create a Swappable Component, you need to use the `createSwappableComponent` function from the `@backstage/frontend-plugin-api` package. You can supply a default implementation for the component, as well as a way to separate both the props of the external component and in the implementation of the component.
+
+```tsx title="in @internal/plugin-example-react"
+import { createSwappableComponent } from '@backstage/frontend-plugin-api';
+
+export const ExampleSwappableComponent = createSwappableComponent({
+  name: 'example',
+
+  // This is a loader for loading the default implementation of the component when there's no overriden
+  // implementation created with `SwappableComponentBlueprint`.
+  // It can be sync like below, but is can also be async like `loader: () => import('./DefaultImplementation').then(m => m.default)`.
+  loader: () => (props: { name: string }) =>
+    <div>Your name is {props.name}</div>,
+
+  // This is an optional function that can be used to transform the props of the external component.
+  // If it's not provided, the props of the external component will be passed through unchanged.
```

Comment:
I think it's good to explain that you typically won't want to define this when first creating the component, but it's instead something that can help keep the external API stable as it evolves